src/Mod/Sketcher/Gui/CommandConstraints.cpp

```diff
@@ -1106,11 +1109,146 @@ class DrawSketchHandlerGenConstraint: public DrawSketchHandler
+            {{QObject::tr("%1 pick first line"), {Gui::InputHint::UserInput::MouseLeft}}}},
+            {"Sketcher_ConstrainAngle", 1,
+            {{QObject::tr("%1 pick second line"), {Gui::InputHint::UserInput::MouseLeft}}}},
+
+            // Symmetry constraint
+            {"Sketcher_ConstrainSymmetric", 0,
+            {{QObject::tr("%1 pick first point"), {Gui::InputHint::UserInput::MouseLeft}}}},
+            {"Sketcher_ConstrainSymmetric", 1,
+            {{QObject::tr("%1 pick second point"), {Gui::InputHint::UserInput::MouseLeft}}}},
+            {"Sketcher_ConstrainSymmetric", 2,
+            {{QObject::tr("%1 pick symmetry line"), {Gui::InputHint::UserInput::MouseLeft}}}},
+
+            // Tangent constraint
+            {"Sketcher_ConstrainTangent", 0,
+            {{QObject::tr("%1 pick first curve or endpoint"), {Gui::InputHint::UserInput::MouseLeft}}}},
+            {"Sketcher_ConstrainTangent", 1,
+            {{QObject::tr("%1 pick second curve or endpoint"), {Gui::InputHint::UserInput::MouseLeft}}}},
+            {"Sketcher_ConstrainTangent", 2,
+            {{QObject::tr("%1 pick point for tangent-via-point"), {Gui::InputHint::UserInput::MouseLeft}}}},
+
+            // Perpendicular constraint
+            {"Sketcher_ConstrainPerpendicular", 0,
+            {{QObject::tr("%1 pick first line or endpoint"), {Gui::InputHint::UserInput::MouseLeft}}}},
+            {"Sketcher_ConstrainPerpendicular", 1,
+            {{QObject::tr("%1 pick second line or endpoint"), {Gui::InputHint::UserInput::MouseLeft}}}},
+            {"Sketcher_ConstrainPerpendicular", 2,
+            {{QObject::tr("%1 pick point for perpendicular-via-point"), {Gui::InputHint::UserInput::MouseLeft}}}},
+
+            // Parallel constraint
+            {"Sketcher_ConstrainParallel", 0,
+            {{QObject::tr("%1 pick first line"), {Gui::InputHint::UserInput::MouseLeft}}}},
+            {"Sketcher_ConstrainParallel", 1,
+            {{QObject::tr("%1 pick second line"), {Gui::InputHint::UserInput::MouseLeft}}}},
+
+            // Distance constraint
+            {"Sketcher_ConstrainDistance", 0,
+            {{QObject::tr("%1 pick first point, line or circle"), {Gui::InputHint::UserInput::MouseLeft}}}},
+            {"Sketcher_ConstrainDistance", 1,
+            {{QObject::tr("%1 pick second point, line or circle"), {Gui::InputHint::UserInput::MouseLeft}}}},
+        };
+
+    }
+
+    static std::list<Gui::InputHint> lookupConstraintHints(const std::string& commandName, int selectionStep) {
+        const auto constraintHintTable = getConstraintHintTable();
+
+        auto it = std::find_if(constraintHintTable.begin(),
+                              constraintHintTable.end(),
+                              [&commandName, selectionStep](const ConstraintHintEntry& entry) {
+                                  return entry.commandName == commandName && entry.selectionStep == selectionStep;
+                              });
```

Comment:
Have you tried [`std::ranges::find_if`](https://en.cppreference.com/w/cpp/algorithm/ranges/find.html) from c++20? it can be used like this (example from PartFeature.cpp):
```c++
    auto foundSimplification =
        std::ranges::find_if(simplificationOrder,
                             [&](TopAbs_ShapeEnum topType) {
                                 return compoundShape.countSubShapes(topType) == 1;
                             });
    if (foundSimplification != simplificationOrder.end()) {
        return compoundShape.getSubTopoShape(*foundSimplification, 1);
    }
```

that way you don't need to specify start end end iterators for the sequence container